### PR TITLE
Populate transaction branch with connection contexts

### PR DESCRIPTION
**Pull request by pcnfernando**
Fixes #[40075](https://github.com/ballerina-platform/ballerina-lang/issues/40075)

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by pcnfernando** [comment]
Full build pipeline triggered in https://github.com/ballerina-platform/ballerina-release/actions/runs/4635308154/jobs/8202249194

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Strand.java [hunk 1]
```diff
@@ -144,6 +144,7 @@ private TransactionLocalContext createTrxContextBranch(TransactionLocalContext c
                         currentTrxContext.getInfoRecord());
         String currentTrxBlockId = currentTrxContext.getCurrentTransactionBlockId();
         trxCtx.addCurrentTransactionBlockId(currentTrxBlockId + "_" + strandName);
+        trxCtx.setTransactionContextStore(currentTrxContext.getTransactionContextStore());
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Strand.java:147]: This method is not calling from all strand constructors. Will that be a problem?

Comment (pcnfernando) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Strand.java:147]: That would be fine. There are two Strand constructs [1]9https://github.com/ballerina-platform/ballerina-lang/blob/cfde8eaf1657f0969eea220d4c6d042b391f0b99/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Strand.java#L94) and [2](https://github.com/ballerina-platform/ballerina-lang/blob/cfde8eaf1657f0969eea220d4c6d042b391f0b99/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Strand.java#L118).
Only [2] creates a new TransactionLocalContext. Other strand just propagates the property bag

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**